File: src/main/java/com/protryon/jasm/NameAndType.java
package com.protryon.jasm;

import com.shapesecurity.functional.data.Either;

public class NameAndType {

    public final String name;
    public final Either<JType, MethodDescriptor> type;

    public NameAndType(String name, Either<JType, MethodDescriptor> type) {
        this.name = name;
        this.type = type;
    }

    public String toString() {
        return this.name + " " + type.either(JType::toString, MethodDescriptor::toString);
    }

}

File: src/main/java/com/protryon/jasm/NameAndType.kt
package com.protryon.jasm

import com.shapesecurity.functional.data.Either

class NameAndType(val name: String, val type: Either<JType, MethodDescriptor>) {

    override fun toString(): String {
        return this.name + " " + type.either({ it.toString() }, { it.toString() })
    }

}
